astronomer/providers/google/cloud/operators/bigquery.py

```diff
@@ -186,17 +186,21 @@ def execute(self, context: Context) -> None:  # noqa: D102
         )
         job = self._submit_job(hook, job_id="")
         context["ti"].xcom_push(key="job_id", value=job.job_id)
-        self.defer(
-            timeout=self.execution_timeout,
-            trigger=BigQueryCheckTrigger(
-                conn_id=self.gcp_conn_id,
-                job_id=job.job_id,
-                project_id=hook.project_id,
-                impersonation_chain=self.impersonation_chain,
-                poll_interval=self.poll_interval,
-            ),
-            method_name="execute_complete",
-        )
+
+        if job.running():
+            self.defer(
+                timeout=self.execution_timeout,
+                trigger=BigQueryCheckTrigger(
+                    conn_id=self.gcp_conn_id,
+                    job_id=job.job_id,
+                    project_id=hook.project_id,
+                    impersonation_chain=self.impersonation_chain,
+                    poll_interval=self.poll_interval,
+                ),
+                method_name="execute_complete",
+            )
+        else:
+            super().execute(context=context)
```

Comment:
same question: https://github.com/astronomer/astronomer-providers/pull/1041#discussion_r1187218915

Reply:
Yes, we're doing the same thing without waiting and retrying, as the job is not in running status. If it is, we'll defer it